Documentation/CodingStyle.md

```diff
@@ -637,3 +637,40 @@ Salt const& m_salt;
 ```cpp
 const Salt& m_salt;
 ```
+
+### Casts
+
+Don't use C-style casts. The C-style cast has [complex behavior](https://en.cppreference.com/w/c/language/cast) that might be undesired in many instances. Be aware of what sort of type conversion the code is trying to achieve, and use the appropriate (!) C++ cast operator, like `static_cast`, `reinterpret_cast`, `bit_cast`, `dynamic_cast` etc.
```

Comment:
I'd want a note in here to first do reasonable legwork to prevent the need for a cast in the first place, and only then reach for the proper cast.

Reply:
Yep; I wrote that thing before our discussion.